src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -82,25 +55,25 @@ private void InitSort(CultureInfo culture)
             fixed (char* pSource = &MemoryMarshal.GetReference(source))
             fixed (char* pValue = &MemoryMarshal.GetReference(value))
             {
+                Debug.Assert(pSource != null);
+                Debug.Assert(pValue != null);
+
                 int ret = Interop.Kernel32.FindStringOrdinal(
                             dwFindStringOrdinalFlags,
                             pSource,
                             source.Length,
                             pValue,
                             value.Length,
                             bIgnoreCase ? 1 : 0);
-                return ret;
-            }
-        }
 
-        internal static int IndexOfOrdinalCore(string source, string value, int startIndex, int count, bool ignoreCase)
-        {
-            Debug.Assert(!GlobalizationMode.Invariant);
-
-            Debug.Assert(source != null);
-            Debug.Assert(value != null);
+                Debug.Assert(ret >= -1 && ret <= source.Length);
+                if (ret < 0 && Marshal.GetLastWin32Error() != Interop.Errors.ERROR_SUCCESS)
```

Comment:
>GetLastWin32Error [](start = 39, length = 17)

I don't like checking the last error. Windows can change this and break us. I would like to depend on what the public doc says and for special cases we can handle it manually.

Reply:
talked offline with Levi and I am fine to have the last error check in the debug build only.